backends/advanced/src/advanced_omi_backend/controllers/system_controller.py

```
@@ -912,3 +919,142 @@ async def save_diarization_settings(settings: dict):
+        await user.save()
+        
+        logger.info(f"Updated primary speakers configuration for user {user.user_id}: {len(primary_speakers)} speakers")
+        
+        return {
+            "message": "Primary speakers configuration updated successfully",
+            "primary_speakers": primary_speakers,
+            "count": len(primary_speakers),
+            "status": "success"
+        }
+        
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)